### PR TITLE
[stable/dex]: Add support to configure nodePort

### PR DESCRIPTION
**Pull request by cjheppell**
**What this PR does / why we need it**:
Adds support for configuring the `nodePort` that will be used when the Dex service type is set to `NodePort`.

### CODE REVIEW

**Comment by unguiculus** [comment]
cc @sstarcher @kfox1111 

BTW, It'd be good if someone could add a readme.

**Comment by kfox1111** [comment]
lgtm

@unguiculus agreed

@unguiculus it would also be good if all the charts could standardize on some library svc implementation. Having every chart go through the "we need nodePort. We need externalIPs. We need, xxx" over and over really isn't productive.

**Comment by cjheppell** [comment]
Thanks for the review, all.

Agreed that a readme should be added, but I feel it's out of scope for this PR.

@unguiculus Is there anything blocking this from being merged that you need me to change?

**Comment by sstarcher** [comment]
lgtm if the readme is updated

**Comment by cjheppell** [comment]
> lgtm if the readme is updated

@sstarcher Not sure what you mean by this, as there currently is no readme?

Unfortunately, I don't have the time right now to go through the chart and create a comprehensive readme of all the possible configuration values. Given that the majority of other chart readme's have been very high quality and useful, the last thing I want to do is try to rush in a readme as part of this PR, which could serve as a potential source of confusion more than anything else. I feel it deserves its own PR where it can be reviewed and critiqued accordingly.

In the meantime, is it possible to get this PR merged? It's currently blocking some work at my organisation (and although we could fork the chart, I'd much prefer to use the current stable release).

Happy to have a discussion on the above points if any disagreement occurs.

File: stable/dex/Chart.yaml
Comment (mattfarina) [inline stable/dex/Chart.yaml:2]: This is being added in a backwards compatible way so you could do a value of `0.2.1` since changes this this are different pre-1.0.0.

File: stable/dex/templates/service.yaml
Comment (mattfarina) [inline stable/dex/templates/service.yaml:23]: If you're looking for a key like this than the values file should have an example that is commented out to direct people. Can you provide that?

**Review by mattfarina — COMMENTED** [review]
I'm ok without the README changes since there is no README. I was kind of surprised to learn that.

File: stable/dex/templates/service.yaml
Comment (cjheppell) [reply stable/dex/templates/service.yaml:23]: Good point.

Comment example added in commit https://github.com/helm/charts/pull/6956/commits/587693b9d2e27419a82b2851457fdcabd269addf

File: stable/dex/Chart.yaml
Comment (cjheppell) [reply stable/dex/Chart.yaml:2]: [This was the case initially](https://github.com/helm/charts/pull/6956/commits/b6a14de0973dc5390742610ab24fc8a308791cc8)

Happy to change it back if you feel it's more representative of the change being made.

**Review by sstarcher — APPROVED** [review]

**Pull request merged**
